### ISSUE TITLE
Selection doesn't work correctly if we start to select the last visible line in the window

### ISSUE THREAD

**Issue by igordmn** [JetBrains/compose-jb#1403]
Compose 1.0.0-beta5, Ubuntu

```
import androidx.compose.foundation.text.selection.SelectionContainer
import androidx.compose.material.Text
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.unit.sp
import androidx.compose.ui.window.singleWindowApplication

fun main() = singleWindowApplication {
    SelectionContainer {
        Text(text = (1..100).joinToString { "Text " }, style = TextStyle(fontSize = 50.sp))
    }
}
```

Start selecting the last line from the center to the left.

Result:
![Screenshot from 2021-11-15 16-23-31](https://user-images.githubusercontent.com/5963351/141789309-96e56507-58cd-42df-b3bc-5e138a0e5c58.png)

The right part of the line is selected, but it shouldn't

**Comment by igordmn** [comment]
Should be fixed in 1.0.0-beta6-dev474

**Comment by igordmn** [comment]
Reopened because the fix was reverted https://github.com/JetBrains/androidx/pull/132

**Comment by igordmn** [comment]
Fixed in 1.0.0-rc5

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-6206">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
